### PR TITLE
add the gitserver API to doc devs

### PR DESCRIPTION
**Pull request by peterguy**
Copied from the Google spreadsheet detailing gitserver and how it translates to other SCMs, like Perforce and SVN. http://go/gitserver-api

Needs a bunch more work to be valid API documentation, like parameters and return values.

## Test plan

docs change

### CODE REVIEW

File: doc/dev/background-information/gitserver-api.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+| AddrForRepo           | resolves repo names to gitserver instances                                                                                                                                                            |                                                                                                                                                                                                                                                                                                                                                                    |                                                                                                                                                                                                                                                                                      |
```
Comment (mrnugget) [inline doc/dev/background-information/gitserver-api.md:12]: Can you format the `Method`s and the `Git Command`s like this? I think that would make it nicer to read.
```suggestion
| `AddrForRepo`           | resolves repo names to gitserver instances                                                                                                                                                            |                                                                                                                                                                                                                                                                                                                                                                    |                                                                                                                                                                                                                                                                                      |
```

File: doc/dev/index.md [hunk 2]
```diff
@@ -108,6 +108,7 @@ Clarification and discussion about key concepts, architecture, and development s
     - [Testing Go code](background-information/languages/testing_go_code.md)
     - [Testing web code](background-information/testing_web_code.md)
 - [Building p4-fusion](background-information/build_p4_fusion.md)
+- [The `gitserver` API](background-information/gitserver-api.md)
```
Comment (mrnugget) [inline doc/dev/index.md:111]: can you also add it to `background-information/index.md`?

**Review by mrnugget — APPROVED** [review]
Very cool!

Left two comments: 

1. formatting
2. missing link

Another thought: should the page maybe have a date on it somewhere? "Last updated: 1 feb 2023" or something. I fear it'll become outdated in next ~6months and I want to avoid possible confusion.

**Review by unknwon — APPROVED** [review]
Thank you! And +1 to Thorsten's comment.

File: doc/dev/background-information/gitserver-api.md (hunk 1 above)
Comment (peterguy) [reply doc/dev/background-information/gitserver-api.md:12]: Great comments and suggestions. I'll make those changes.

**Pull request merged**